src/com/android/volley/RequestQueue.java

```diff
@@ -301,7 +295,16 @@ public boolean isProcessing(Request< ? > request) {
                 mWaitingRequests.put(cacheKey, null);
             }
 
-            mCacheQueue.add(request);
+            if (!request.isJoined()) {
```

Comment:
the request can join if it is network only now. This connects it to an existing request of the same type.